src/dmd/json.d

```diff
@@ -826,8 +826,60 @@ public:
     private void generateCompilerInfo()
     {
         objectStart();
-        property("binary", global.params.argv0);
+        property("vendor", global.compiler.vendor);
         property("version", global._version);
+        property("argv0", global.params.argv0);
```

Comment:
Well, I was thinking that `argv0` is more true to what it actually is.  "binary" sounds more like a filename but argv0 can be an alias/string/filename.  It may be helpful to include both "binary" and "argv0"...